tests/test_kernels.py

```diff
@@ -17,57 +21,67 @@
+    GridType.IRREGULAR_WITH_LAND: [
+        "wet_mask",
+        "dxw",
+        "dyw",
+        "dxs",
+        "dys",
+        "area",
+        "kappa_w",
+        "kappa_s",
+    ],
+    GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED: ["wet_mask", "area"],
+    GridType.TRIPOLAR_POP_WITH_LAND: ["wet_mask", "dxe", "dye", "dxn", "dyn", "tarea"],
+}
+
+
+def _make_random_data(shape: Tuple[int, int]) -> np.ndarray:
+    return rng.random(shape)
+
+
+def _make_mask_data(shape: Tuple[int, int]) -> np.ndarray:
+    mask_data = np.ones(shape)
+    ny, nx = shape
+    mask_data[0, :] = 0  #  Antarctica; required for some kernels
+    mask_data[: (ny // 2), : (nx // 2)] = 0
+    return mask_data
+
+
+def _make_irregular_grid_data(shape: Tuple[int, int]) -> np.ndarray:
+    return 0.5 + rng.random(shape)
+
+
 ################## Scalar Laplacian tests ##############################################
 @pytest.fixture(scope="module", params=scalar_grids)
 def grid_type_field_and_extra_kwargs(request):
     grid_type = request.param
-    ny, nx = (128, 256)
-    data = np.random.rand(ny, nx)
+    shape = (128, 256)
+
+    data = _make_random_data(shape)
 
     extra_kwargs = {}
-    if grid_type == GridType.REGULAR_AREA_WEIGHTED:
-        area = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["area"] = area
-    if grid_type == GridType.REGULAR_WITH_LAND:
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        extra_kwargs["wet_mask"] = mask_data
-    if grid_type == GridType.REGULAR_WITH_LAND_AREA_WEIGHTED:
-        area = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["area"] = area
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        extra_kwargs["wet_mask"] = mask_data
-    if grid_type == GridType.IRREGULAR_WITH_LAND:
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        extra_kwargs["wet_mask"] = mask_data
-        grid_data = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["dxw"] = grid_data
-        extra_kwargs["dyw"] = grid_data
-        extra_kwargs["dxs"] = grid_data
-        extra_kwargs["dys"] = grid_data
-        extra_kwargs["area"] = grid_data * grid_data
-        extra_kwargs["kappa_w"] = np.ones_like(data)
-        extra_kwargs["kappa_s"] = np.ones_like(data)
-    if grid_type == GridType.TRIPOLAR_REGULAR_WITH_LAND_AREA_WEIGHTED:
-        area = 0.5 + np.random.rand(ny, nx)
-        extra_kwargs["area"] = area
-        mask_data = np.ones_like(data)
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        mask_data[0, :] = 0  #  Antarctica
-        extra_kwargs["wet_mask"] = mask_data
+    for name in _grid_kwargs[grid_type]:
+        if name == "wet_mask":
+            extra_kwargs[name] = _make_mask_data(shape)
+        elif "kappa" in name:
+            extra_kwargs[name] = np.ones(shape)
+        else:
+            extra_kwargs[name] = _make_irregular_grid_data(shape)
```

Comment:
Note here that all of the irregular grid variables (e.g. dx, dy, etc.) are now getting distinct random data.

Reply:
This is great. Previously, I had assigned identical random data to `dx` and `dy`, which made the conservation test for `TRIPOLAR_POP_WITH_LAND` accidentally pass (see *Side note for the curious* in https://github.com/ocean-eddy-cpt/gcm-filters/pull/80#issue-701806629).